lib/src/util/element_type.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:react/react_client/react_interop.dart';
+
+/// A class used to strongly type MUI's string shorthand for React components
+class ElementType {
+  dynamic _element;
+  dynamic get value => _element;
+
+  ElementType(this._element);
+
+  /// Create an ElementType from a string
+  /// element can be any HTML element as a string, i.e. 'div'
+  ElementType.fromString(String element) : _element = element;
+
+  /// Create an ElementType from a ReactComponent
+  ElementType.fromComponent(ReactComponent element) : _element = element;
```

Comment:
`ReactComponent` is actually the JS interop type for a component instance for class-based component, analogous to the JS `React.Component` class.

Here, we want the variable to represent the component type itself, and not an instance of that type. `ReactClass` would be a better type for this.

However, in addition to class components, component types can also be function components (or as you've implemented above, strings for DOM types), so `dynamic` might be the best type for this.

```suggestion
  ElementType.fromType(dynamic type) : _element = type;
```

All `ReactComponentFactoryProxy`s have a `type` field which maps to this type, so it might be convenient to expose that as a constructor, in addition to  one for `UiFactory`:
```dart
  ElementType.fromFactory(ReactComponentFactory factory) : _element = factory.type;
  ElementType.fromUiFactory(UiFactory factory) : this.fromFactory(factory().componentFactory);

```

So then you could do things like:
```dart
UiFactory<FooProps> Foo = uiFunction((props) {
  // Component impl ..
}, ... );
```
```dart
ElementType.fromUiFactory(Foo)
```

Reply:
This is really helpful, thanks Greg!

Reply:
Updated in future commit

Reply:
Sure thing!